File: Lucidity/FIELDForm.vb
Imports System.IO
Imports System.Xml

Public Class FIELDForm

    Private m_strPath As String = DataDirectory.GetFolderPath() + "\Lightened Dream\"
    Private m_boolMouseDown As Boolean
    Private m_boolStartTime As DateTime

    Private Sub FIELDForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ' Check if the FIELD file exists
        If (File.Exists(m_strPath & "FIELD.ld3")) Then
            Dim xmlDocFIELD As New XmlDocument
            xmlDocFIELD.Load(m_strPath & "FIELD.ld3")
            txtAnthem.Text = xmlDocFIELD.DocumentElement.SelectSingleNode("Anthem").InnerText
        End If
        If txtAnthem.Text = "" Then
            txtAnthem.Text = m_strPath & "Anthem.mp3"
        End If
    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        Close()
    End Sub

    Private Sub btnOk_MouseDown(ByVal sender As System.Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles btnOk.MouseDown
        m_boolMouseDown = True
        m_boolStartTime = DateTime.Now
    End Sub

    Private Sub btnOk_MouseUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.MouseEventArgs) Handles btnOk.MouseUp
        m_boolMouseDown = False

        Try

            ' Play the anthem
            If (File.Exists(txtAnthem.Text)) Then

                objPlayer.close()
                Application.DoEvents()
                Me.objPlayer.settings.volume = 100
                objPlayer.URL = txtAnthem.Text

            End If

        Catch ex As Exception
            MessageBox.Show(ex.Message, "LightenedDream.FIELD.Induce()", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub

    Private Sub tmrFIELD_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tmrFIELD.Tick
        If m_boolMouseDown Then
            Dim dtDiff As TimeSpan = DateTime.Now.Subtract(m_boolStartTime)
            Dim strDiff As String = ""

            If dtDiff.Days < 10 Then
                strDiff &= "0" & dtDiff.Days
            Else
                strDiff &= dtDiff.Days
            End If
            If dtDiff.Hours < 10 Then
                strDiff &= ":" & "0" & dtDiff.Hours
            Else
                strDiff &= ":" & dtDiff.Hours
            End If
            If dtDiff.Minutes < 10 Then
                strDiff &= ":" & "0" & dtDiff.Minutes
            Else
                strDiff &= ":" & dtDiff.Minutes
            End If
            If dtDiff.Seconds < 10 Then
                strDiff &= ":" & "0" & dtDiff.Seconds
            Else
                strDiff &= ":" & dtDiff.Seconds
            End If
            If dtDiff.Milliseconds < 10 Then
                strDiff &= ":" & "0" & dtDiff.Milliseconds
            Else
                strDiff &= ":" & dtDiff.Milliseconds
            End If

            Text = "FIELD Inducer - " & strDiff
        End If
    End Sub

    Private Sub btnBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBrowse.Click
        Try
            If objOpenFileDialog.ShowDialog = DialogResult.OK Then
                txtAnthem.Text = objOpenFileDialog.FileName
            End If

            ' Save the FIELD file
            Dim strXml As String
            strXml = "<LightenedDream>"
            strXml += "<Anthem>"
            strXml += "</Anthem>"
            strXml += "</LightenedDream>"
            Dim xmlDocFIELD As New XmlDocument
            xmlDocFIELD.LoadXml(strXml)
            xmlDocFIELD.DocumentElement.SelectSingleNode("Anthem").InnerText = txtAnthem.Text
            xmlDocFIELD.Save(m_strPath & "FIELD.ld3")
        Catch ex As Exception
            MessageBox.Show(ex.Message, "LightenedDream.FIELD.Save()", MessageBoxButtons.OK, MessageBoxIcon.Error)
        End Try
    End Sub
End Class

File: Lucidity/FIELDForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.IO;
using System.Xml;
namespace lucidcode.LightenedDream.Lucidity
{

	public partial class FIELDForm
	{

		private string m_strPath = DataDirectory.GetFolderPath() + "\\Lightened Dream\\";
		private bool m_boolMouseDown;

		private DateTime m_boolStartTime;
		private void FIELDForm_Load(System.Object sender, System.EventArgs e)
		{
			// Check if the FIELD file exists
			if ((File.Exists(m_strPath + "FIELD.ld3"))) {
				XmlDocument xmlDocFIELD = new XmlDocument();
				xmlDocFIELD.Load(m_strPath + "FIELD.ld3");
				txtAnthem.Text = xmlDocFIELD.DocumentElement.SelectSingleNode("Anthem").InnerText;
			}
			if (string.IsNullOrEmpty(txtAnthem.Text)) {
				txtAnthem.Text = m_strPath + "Anthem.mp3";
			}
		}

		private void btnCancel_Click(System.Object sender, System.EventArgs e)
		{
			Close();
		}

		private void btnOk_MouseDown(System.Object sender, System.Windows.Forms.MouseEventArgs e)
		{
			m_boolMouseDown = true;
			m_boolStartTime = DateTime.Now;
		}

		private void btnOk_MouseUp(System.Object sender, System.Windows.Forms.MouseEventArgs e)
		{
			m_boolMouseDown = false;


			try {
				// Play the anthem

				if ((File.Exists(txtAnthem.Text))) {
					objPlayer.close();
					Application.DoEvents();
					this.objPlayer.settings.volume = 100;
					objPlayer.URL = txtAnthem.Text;

				}

			} catch (Exception ex) {
				MessageBox.Show(ex.Message, "LightenedDream.FIELD.Induce()", MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}

		private void tmrFIELD_Tick(System.Object sender, System.EventArgs e)
		{
			if (m_boolMouseDown) {
				TimeSpan dtDiff = DateTime.Now.Subtract(m_boolStartTime);
				string strDiff = "";

				if (dtDiff.Days < 10) {
					strDiff += "0" + dtDiff.Days;
				} else {
					strDiff += dtDiff.Days;
				}
				if (dtDiff.Hours < 10) {
					strDiff += ":" + "0" + dtDiff.Hours;
				} else {
					strDiff += ":" + dtDiff.Hours;
				}
				if (dtDiff.Minutes < 10) {
					strDiff += ":" + "0" + dtDiff.Minutes;
				} else {
					strDiff += ":" + dtDiff.Minutes;
				}
				if (dtDiff.Seconds < 10) {
					strDiff += ":" + "0" + dtDiff.Seconds;
				} else {
					strDiff += ":" + dtDiff.Seconds;
				}
				if (dtDiff.Milliseconds < 10) {
					strDiff += ":" + "0" + dtDiff.Milliseconds;
				} else {
					strDiff += ":" + dtDiff.Milliseconds;
				}

				Text = "FIELD Inducer - " + strDiff;
			}
		}

		private void btnBrowse_Click(System.Object sender, System.EventArgs e)
		{
			try {
				if (objOpenFileDialog.ShowDialog() == DialogResult.OK) {
					txtAnthem.Text = objOpenFileDialog.FileName;
				}

				// Save the FIELD file
				string strXml = null;
				strXml = "<LightenedDream>";
				strXml += "<Anthem>";
				strXml += "</Anthem>";
				strXml += "</LightenedDream>";
				XmlDocument xmlDocFIELD = new XmlDocument();
				xmlDocFIELD.LoadXml(strXml);
				xmlDocFIELD.DocumentElement.SelectSingleNode("Anthem").InnerText = txtAnthem.Text;
				xmlDocFIELD.Save(m_strPath + "FIELD.ld3");
			} catch (Exception ex) {
				MessageBox.Show(ex.Message, "LightenedDream.FIELD.Save()", MessageBoxButtons.OK, MessageBoxIcon.Error);
			}
		}
		public FIELDForm()
		{
			Load += FIELDForm_Load;
			InitializeComponent();
		}
	}
}
